// @redwoodjs/cli/dist/commands/experimental/setupInngest.js
"use strict";

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js/object/define-property");
var _interopRequireDefault = require("@babel/runtime-corejs3/helpers/interopRequireDefault").default;
_Object$defineProperty(exports, "__esModule", {
  value: true
});
exports.handler = exports.description = exports.command = exports.builder = exports.EXPERIMENTAL_TOPIC_ID = void 0;
var _promise = _interopRequireDefault(require("@babel/runtime-corejs3/core-js/promise"));
var _interopRequireWildcard2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/interopRequireWildcard"));
var _util = require("./util");
const command = 'setup-inngest';
exports.command = command;
const description = 'Setup Inngest for background, scheduled, delayed, multi-step, and fan-out jobs';
exports.description = description;
const EXPERIMENTAL_TOPIC_ID = 4866;
exports.EXPERIMENTAL_TOPIC_ID = EXPERIMENTAL_TOPIC_ID;
const builder = yargs => {
  yargs.option('force', {
    alias: 'f',
    default: false,
    description: 'Overwrite existing configuration',
    type: 'boolean'
  }).epilogue((0, _util.getEpilogue)(command, description, EXPERIMENTAL_TOPIC_ID, true));
};
exports.builder = builder;
const handler = async options => {
  const {
    handler
  } = await _promise.default.resolve().then(() => (0, _interopRequireWildcard2.default)(require('./setupInngestHandler')));
  return handler(options);
};
exports.handler = handler;
